app/src/main/java/com/github/factotum_sdp/factotum/ui/bossmap/BossMapFragment.kt

```diff
@@ -0,0 +1,184 @@
+package com.github.factotum_sdp.factotum.ui.bossmap
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import com.github.factotum_sdp.factotum.R
+import com.github.factotum_sdp.factotum.hasLocationPermission
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.MapView
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MapStyleOptions
+import com.google.android.gms.maps.model.MarkerOptions
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.DatabaseReference
+import com.google.firebase.database.FirebaseDatabase
+import com.google.firebase.database.ValueEventListener
+
+private const val ZOOM_LEVEL_CITY = 14f
+private const val WAIT_TIME_LOCATION_UPDATE = 30000L
+
+class BossMapFragment : Fragment(), OnMapReadyCallback {
+
+    private lateinit var mapView: MapView
+    private lateinit var googleMap: GoogleMap
+    private lateinit var database: DatabaseReference
+    private lateinit var fusedLocationClient: FusedLocationProviderClient
+    private val handler = Handler(Looper.getMainLooper())
+    private val requestPermissionLauncher = registerForActivityResult(
+        ActivityResultContracts.RequestPermission()
+    ) { isGranted ->
+        if (isGranted) {
+            requireContext().hasLocationPermission()
+            activateLocation(googleMap)
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_boss_map, container, false)
+        mapView = view.findViewById(R.id.mapView)
+        mapView.onCreate(savedInstanceState)
+        mapView.getMapAsync(this)
+
+        // Initialize Firebase database
+        database = FirebaseDatabase.getInstance().getReference("Location")
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(requireActivity())
```

Comment:
It is not needed to launch a second FusedLocationProvider inside the app.
Take the observable in the UserViewModel, you can easily fetch them from here. 

If the live location is not set, a solution would be to use the couriers locations to set the zoom and the camera position, otherwise a default location.